sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryErrorsBase.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.errors
+
+import org.apache.spark.sql.catalyst.expressions.Literal
+import org.apache.spark.sql.types.StringType
+import org.apache.spark.unsafe.types.UTF8String
+
+trait QueryErrorsBase {
+  // Converts an error class parameter to its SQL representation
+  def toSQLValue(v: Any): String = v match {
+    case _: UTF8String => Literal.create(v, StringType).sql
+    case _ => Literal(v).sql
```

Comment:
Checked the [Literal.sql](https://github.com/apache/spark/blob/v3.2.1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala#L454) function, for some data types the sql string may be not as expected to be  shown in error msg, for example `null` values and `FloatType` values, the sql string would be a `cast` expression.

```
    // Float type doesn't have a suffix
    case (v: Float, FloatType) =>
      val castedValue = v match {
        case _ if v.isNaN => "'NaN'"
        case Float.PositiveInfinity => "'Infinity'"
        case Float.NegativeInfinity => "'-Infinity'"
        case _ => s"'$v'"
      }
      s"CAST($castedValue AS ${FloatType.sql})"
```

Reply:
> the sql string may be not as expected

What is the expected from your point of view? Output of error messages should be consistent with other places, if we output a cast in other places, ok, let's do that in errors as well.

Reply:
I agree it's weird to see an unrelated `CAST` in the error message. Can we add some special cases here? e.g. null value is `NULL`, float value should be the same as the double value.

Reply:
Extended the `toSQLValue` function to handle the special cases.